dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+        if isinstance(kibana_objects_file, IOBase):
+            reqdata = {'file': ('dynamite_import.ndjson', kibana_objects_file)}
+        else:
+            reqdata = {'file': kibana_objects_file}
+        resp = requests.post(url, params=params, auth=auth, files=reqdata, headers={'kbn-xsrf': 'true'})
+        if resp.status_code not in range(200, 299):
+            self.logger.error(f'Kibana endpoint returned a {resp.status_code} - {resp.text}')
+            # TODO raise exception
+        return resp.json()
+
+    def install(self, package_install_path: str, username: Optional[str] = None, password: Optional[str] = None):
+        """Install a package. A package can be given as an archive or directory. A package must contain one or more ndjson files and a manifest.json
+
+        Args:
+            package_install_path (str): path to the file or folder of files
+            username (Optional[str], optional): kibana auth usrname. Defaults to None.
+            password (Optional[str], optional): kibana auth password. Defaults to None.
+        """
+        is_folder = os.path.isdir(package_install_path)
+        filepaths = []
+        if not is_folder:
+            filepaths = [package_install_path]
+        else:
+            # check for installable items by extension, they will be verified by mimetype later.
+            acceptable_extensions = ['tar.xz', 'tar.gz']
+            for itm in os.listdir(package_install_path):
+                for ex in acceptable_extensions:
+                    if itm.endswith(ex):
+                        filepaths.append(f"{package_install_path}{itm}")
+                        break
+            print(f"Found {len(filepaths)} packages to install.")
+        for _filepath in filepaths:
+            filepath = os.path.abspath(_filepath)
+            # check mimetype of the file to determine how to proceed
+            filetype, encoding = mimetypes.MimeTypes().guess_type(filepath)
+            installation_statuses = []
+            # default to orphan package in case of install from .ndjson
+            manifest = PackageManifest(OrphanPackageData)
+            package = Package(manifest)
+            print(filepath)
+            if filetype == 'application/x-tar' or encoding == 'gzip':
+                # handle tarfile
+                tar = tarfile.open(filepath)
+                try:
+                    manifest = tar.extractfile('manifest.json')
+                    manifest = PackageManifest(manifest.read().decode('utf8'))
+                    package = Package(manifest)
+                    package.create_packages_index()
+                    # check if package exists already.
+                    existing = Package.find_by_slug(package.manifest.create_slug())
+                    if existing:
+                        rmexisting = input(f"A Package titled {existing.manifest.name} is already installed, do you want to uninstall it? [y/n]") in "yY"
```

Comment:
The user should be able to bypass this prompt. Please provide a boolean in this method that can be converted to commandline flag (E.G ignore_warnings).